packages/frontity/src/commands/subscribe.ts

```diff
@@ -1,19 +1,17 @@
-import { EventEmitter } from "events";
 import ora from "ora";
 import chalk from "chalk";
 import subscribe from "../steps/subscribe";
 import { errorLogger } from "../utils";
+import { emitter } from "../utils/eventEmitter";
 
 export default async (email: string) => {
-  const emitter = new EventEmitter();
-
-  emitter.on("error", errorLogger);
-  emitter.on("subscribe", (message, action) => {
+  emitter.on("cli:subscribe:error", errorLogger);
+  emitter.on("cli:subscribe", (message, action) => {
     if (action) ora.promise(action, message);
     else console.log(message);
   });
 
-  await subscribe(email, emitter);
+  await subscribe(email);
```

Comment:
I know you suggested using `event-promised` for this but for the reasons that I've outlined here: https://community.frontity.org/t/expose-frontity-commands-to-be-used-programmatically/977/6?u=mmczaplinski using a promise will not work. 

Unless you had a different idea in mind that I did not understand, of course 😅 

My thinking was the following: 
1. A command calls some step.
2. A step performs some work, whatever it needs to do.
3. A step finished work and wants to log a message to the user in the terminal. For example to let them know that they subscribed successfully
4. A step should not `console.log` anything directly, instead, it will emit an event. For example: "cli:user:subscribed"
5. The event will be caught in the interface and the interface will print the message on the terminal

Reply:
Here I'm talking about steps, not commands.

In my opinion, steps should return simple promises, not event emitters because they should either succeed or fail. It should not be a concern of the step to log a message (or to have the intent of logging a message).

Reply:
Oh right, okay, this makes sense. As long as the above assumption is true
of course (that steps always only succeed or fail and don't return any
information back to the command).

Should we just try it?

Reply:
Sure, go ahead and let's see what happens! :)

By the way, if they succeed, they can return a message using the result of the promise. And if they fail, they can return a message with information about the error. 

```js
try {
  const availablePort = await checkForAvailablePort({ default: 3000 });
  // => Maybe availablePort is 3001 because 3000 is used
} catch (error) {
  console.error(error.message);
  throw new Error(error);
  // => Maybe error.message is "You are not going to believe this but
  // there's not a single available port in your computer! 😱".
}
```